src/pudl/analysis/allocate_net_gen.py

```diff
@@ -1,103 +1,100 @@
-  :func:`aggregate_gen_fuel_by_generator`.
-
-MOAR High-level description about the allocaiton step:
-
-We allocate the net generation/fuel consumption reported in the
-generation_fuel_eia923/boiler_fuel_eia923 table on the basis of plant, prime mover,
-and fuel type among the generators in each plant that have matching fuel types.
-Generation/fuel consumption is allocated proportional to reported generation/fuel
-consumption if it's available, and proportional to each generator's capacity if
-generation/fuel consumption is not available.
+#. **Read inputs**: Read denormalized net generation and fuel consumption data from the
+   PUDL DB (see :func:`extract_input_tables`). Standardize reporting frequency (see
+   :func:`standardize_input_frequency`) which employs
+   :func:`distribute_annually_reported_data_to_months_if_annual` and
+   :func:`pudl.helpers.expand_timeseries`.
+#. **Associate inputs**: Merge the data columns from the all of the input tables as
+   long as they share generator id, prime mover code and/or energy source codes. The
+   output of this step is a table with the primary keys py:const:`IDX_GENS_PM_ESC` with
+   the data columns from the input tables broadcasted (see
+   :func:`associate_generator_tables`).
+#. **Flag associations**: Prepare for the allocation by adding boolean flags for whether
+   a particular :py:const:`IDX_GENS_PM_ESC` shows up in each of the input data tables.
+   This enables us to determine what kind of coverage the granular tables have and from
+   that how the record should be allocated (see :func:`prep_alloction_fraction`).
+#. **Allocate**: Allocate the net generation and fuel from the less granular
+   :ref:`generation_fuel_eia923` table to the :py:const:`IDX_GENS_PM_ESC` level. More
+   details on the allocation process below (see :func:`allocate_net_gen_by_gen_esc` and
+   :func:`allocate_fuel_by_gen_esc`).
+#. **Sanity check allocation**: Verify that the total allocated net generation and fuel
+   consumption within each plant is equal to the total of the originally reported values
+   within some tolerance (see :func:`test_original_gf_vs_the_allocated_by_gens_gf`).
+   Warn if assumptions about the data and the outputs aren't met (see
+   :func:`warn_if_missing_pms`, :func:`_test_frac`, :func:`test_gen_fuel_allocation` and
+   :func:`_test_gen_pm_fuel_output`)
+#. **Aggregate output**: Aggregate the main output (which has primary keys of
+   :py:const:`IDX_GENS_PM_ESC`) to the generator-level with primary keys of
+   :py:const:`IDX_GENS` (see :func:`aggregate_gen_fuel_by_generator`).
+
+**High-level description about the allocaiton step**:
+
+We allocate the data columns reported in the :ref:`generation_fuel_eia923` table on the
+basis of plant, prime mover, and energy source among the generators in each plant that
+have matching energy sources.
 
 In more detail, within each reporting period, we split the plants into three groups:
```

Comment:
i am unsure this is totally correct. i'm going to get through the rest of the comments and get back to this one!

Reply:
```
We group the associated data columns by :py:const:`IDX_PM_ESC` and categorize
each resulting group of generators based on whether  **ALL**, **SOME**, or **NONE**
of them reported data in the granular tables. This is done for both the net generation and
fuel consumption since the same generator may have reported differently in its respective
granular table.
```